apps/data_handler/handlers/events/nostra/transform_events.py

```diff
@@ -0,0 +1,252 @@
+)
+from data_handler.handler_tools.api_connector import DeRiskAPIConnector
+from data_handler.handler_tools.constants import ProtocolAddresses
+from data_handler.handler_tools.data_parser.nostra import NostraDataParser
+from data_handler.handler_tools.data_parser.serializers import (
+    BearingCollateralBurnEventData,
+    BearingCollateralMintEventData,
+    DebtBurnEventData,
+    DebtMintEventData,
+    DebtTransferEventData,
+    InterestRateModelEventData,
+)
+from shared.constants import ProtocolIDs
+
+logger = logging.getLogger(__name__)
+
+EVENT_MAPPING: Dict[str, Tuple[Callable, str]] = {
+    "DebtMint": (NostraDataParser.parse_debt_mint_event, "save_debt_mint_event"),
+    "DebtBurn": (NostraDataParser.parse_debt_burn_event, "save_debt_burn_event"),
+    "DebtTransfer": (
+        NostraDataParser.parse_debt_transfer_event,
+        "save_debt_transfer_event",
+    ),
+    "InterestRateModelUpdate": (
+        NostraDataParser.parse_interest_rate_model_event,
+        "save_interest_rate_model_event",
+    ),
+    "BearingCollateralMint": (
+        NostraDataParser.parse_interest_bearing_collateral_mint_event,
+        "save_bearing_collateral_mint_event",
+    ),
+    "BearingCollateralBurn": (
+        NostraDataParser.parse_interest_bearing_collateral_burn_event,
+        "save_bearing_collateral_burn_event",
+    ),
+}
+
+
+class NostraTransformer:
+    """
+    A class that is used to transform Nostra events into database models.
+    """
+
+    EVENT_MAPPING: Dict[str, Tuple[Callable, str]] = EVENT_MAPPING
+    PROTOCOL_ADDRESSES: str = ProtocolAddresses.NOSTRA_MARKET_ADDRESSES
+    PROTOCOL_TYPE: ProtocolIDs = ProtocolIDs.NOSTRA_ALPHA
+    PAGINATION_SIZE: int = 1000
+
+    def __init__(self):
+        """
+        Initialize the NostraTransformer instance.
+        Initializes API and database connectors, and retrieves the last processed block number.
+        """
+        self.api_connector = DeRiskAPIConnector()
+        self.db_connector = NostraEventDBConnector()
+        self.last_block = self.db_connector.get_last_block(self.PROTOCOL_TYPE)
+
+    def fetch_and_transform_events(
+        self, from_address: str, min_block: int, max_block: int
+    ) -> None:
+        """
+        Fetch events from the DeRisk API and transform them into database models.
+        """
+        # Fetch events using the API connector
+        response = self.api_connector.get_data(
+            from_address=from_address,
+            min_block_number=min_block,
+            max_block_number=max_block,
+        )
+
+        if "error" in response:
+            raise ValueError(f"Error fetching events: {response['error']}")
+
+        # Process each event based on its type
+        for event in response:
+            event_type = event.get("key_name")
+            if event_type in self.EVENT_MAPPING:
+                parser_func, save_to_db_method_name = self.EVENT_MAPPING[event_type]
+
+                if event_type == "DebtTransfer":
```

Comment:
Please, we will have more than 2 events, so do it that way that we can scale it with other events.
I'd recommend add mapping by event_type